### PR TITLE
[Wallet] Move IncOrderPosNext to CWallet::AddAccountingEntry

### PR DESCRIPTION
**Pull request by pstratem**

### CODE REVIEW

**Comment by jonasschnelli** [comment]
Concept ACK.
Accounting tests (unit tests) are failing.

File: src/wallet/wallet.h [hunk 1]
```diff
@@ -737,7 +739,6 @@ class CWallet : public CCryptoKeyStore, public CValidationInterface
      * Increment the next transaction order id
```
Comment (MarcoFalke) [inline src/wallet/wallet.h]: Could you move the comment as well?

**Comment by BangkitSedar** [comment]
Try new pull

**Comment by pstratem** [comment]
replaced by #8696

**Pull request closed without merging**